src/States/PlanetSelectState.cpp

```diff
@@ -36,8 +36,14 @@ class Explosion
 			mTimer.reset();
 		}
 
-		r.drawSubImageRotated(mSheet, x, y, (mFrame % 8) * 128, (mFrame / 8) * 128, 128, 128, 270.0f);
-		//r.drawSubImage(mSheet, x, y, (mFrame % 8) * 128, (mFrame / 8) * 128, 128, 128);
+		auto rasterX = static_cast<float>(x);
+		auto rasterY = static_cast<float>(y);
+		auto posX = static_cast<float>((mFrame % 8u) * 128.0f);
+		auto posY = static_cast<float>((mFrame / 8u) * 128.0f);
```

Comment:
I wonder if the scope of these `static_casts` are perhaps not quite right. The inner expression appears to be an `unsigned int` `*` `float`. Maybe the cast should be wrapping the `unsigned int` portion. Alternatively, remove the `.0f` from the `128.0f`, and maybe the `u` as well, and just do the whole internal calculation as an integer.

Reply:
I have no idea why I did that. With proper use of parenthesis Integer-to-float promotion should take over: `auto posX = (mFrame % 8u) * 128.0f` and `auto posY = (mFrame / 8u) * 128.0f`

Reply:
Fixed